### PR TITLE
Develop h

### PR DESCRIPTION
**Pull request by vladyslav-harkusha**
[DEMO LINK](https://vladyslav-harkusha.github.io/react_todo-app-with-api/)

### CODE REVIEW

**Review by xiimiik — APPROVED** [review]
LGTM

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]
Check please the functionality for todo renaming, it doesn't work

File: src/App.tsx [hunk 1]
```diff
@@ -1,223 +1,235 @@
...
+      if (user) {
+        const responseTodos = await getTodos(user.id);
+
+        setTodos(responseTodos);
+      }
+    } catch (error) {
+      setIsError(true);
+      setErrorText('Unable to load todos');
+
+      throw new Error(`unexpected error with loading todos: ${error}`);
```
Comment (Anna-Tkachenko) [inline src/App.tsx:54]: you're already handle the error via error and error text variables, throwing error here is redundant (you also didn't cover loadTodos call in try catch functionality)

File: src/api/todos.ts [hunk 2]
```diff
@@ -1,8 +1,19 @@
 import { Todo } from '../types/Todo';
+import { TodoToAdd } from '../types/TodoToAdd';
 import { client } from '../utils/fetchClient';
 
 export const getTodos = (userId: number) => {
   return client.get<Todo[]>(`/todos?userId=${userId}`);
 };
 
-// Add more methods here
+export const addTodo = (newTodo: TodoToAdd) => {
```
Comment (Anna-Tkachenko) [inline src/api/todos.ts:9]: will be great to add types for return values for api functions

**Review by Anna-Tkachenko — APPROVED** [review]
Good job! ⭐️

**Pull request closed without merging**